src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -216,7 +211,7 @@ public void Encode(RlpStream rlpStream, BlockHeader? header, RlpBehaviors rlpBeh
                 }
             }
 
-            if (header.Number >= Eip1559TransitionBlock)
+            if (header.BaseFeePerGas != 0)
```

Comment:
can base fee be 0 in normal cases ? or is it always a non-zero value ? if the former maybe using nullables to indicate the value's presence would be better

Reply:
It can't be zero. Initial plan was to make it nullable, but in lots of places code expects zero as baseFee so it feels safer. And also not-nullable solution breaks a lot less tests